google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITObjectTest.java

```diff
@@ -546,49 +497,76 @@ public void testListBlobsWithOffset() throws ExecutionException, InterruptedExce
     }
   }
 
-  @Test(timeout = 5000)
-  // This test is currently timing out for GRPC
-  @CrossRun.Exclude(transports = Transport.GRPC)
-  public void testListBlobsCurrentDirectory() throws InterruptedException {
-
-    String directoryName = "test-list-blobs-current-directory/";
-    String subdirectoryName = "subdirectory/";
-    String[] blobNames = {directoryName + subdirectoryName + "blob1", directoryName + "blob2"};
-    BlobInfo blob1 = BlobInfo.newBuilder(bucket, blobNames[0]).setContentType(CONTENT_TYPE).build();
-    BlobInfo blob2 = BlobInfo.newBuilder(bucket, blobNames[1]).setContentType(CONTENT_TYPE).build();
-    Blob remoteBlob1 = storage.create(blob1, BLOB_BYTE_CONTENT);
-    Blob remoteBlob2 = storage.create(blob2, BLOB_BYTE_CONTENT);
-    assertNotNull(remoteBlob1);
-    assertNotNull(remoteBlob2);
-    Page<Blob> page =
+  @Test
+  public void testListBlobsCurrentDirectoryIncludesBothObjectsAndSyntheticDirectories() {
+    String bucketName = bucket.getName();
+    String directoryName = "test-list-blobs-current-directory";
+    String subdirectoryName = "subdirectory";
+
+    String uriSubDir = String.format("gs://%s/%s/%s/", bucketName, directoryName, subdirectoryName);
+    String uri1 = String.format("gs://%s/%s/%s/blob1", bucketName, directoryName, subdirectoryName);
+    String uri2 = String.format("gs://%s/%s/blob2", bucketName, directoryName);
+
+    BlobId id1 = BlobId.fromGsUtilUri(uri1);
+    BlobId id2 = BlobId.fromGsUtilUri(uri2);
+    BlobId idSubDir = BlobId.fromGsUtilUri(uriSubDir);
+
+    BlobInfo blob1 = BlobInfo.newBuilder(id1).build();
+    BlobInfo blob2 = BlobInfo.newBuilder(id2).build();
+    BlobInfo obj1Gen1 = storage.create(blob1, BLOB_BYTE_CONTENT).asBlobInfo();
+    BlobInfo obj2Gen1 = storage.create(blob2, BLOB_BYTE_CONTENT).asBlobInfo();
+
+    Page<Blob> page1 =
         storage.list(
-            bucket.getName(),
-            Storage.BlobListOption.prefix("test-list-blobs-current-directory/"),
-            Storage.BlobListOption.currentDirectory());
-    // Listing blobs is eventually consistent, we loop until the list is of the expected size. The
-    // test fails if timeout is reached.
```

Comment:
Listing objects is strongly consistent[1], we don't need the manual retry loop here.

[1] https://cloud.google.com/storage/docs/consistency#strongly_consistent_operations